lib/cisco_node_utils/bgp.rb

```diff
@@ -338,23 +340,35 @@ def default_enforce_first_as
 
     # Confederation Peers (Getter/Setter/Default)
     def confederation_peers
-      config_get('bgp', 'confederation_peers', @get_args)
-    end
-
-    def confederation_peers_set(peers)
-      # The confederation peers command is additive so we first need to
-      # remove any existing peers.
-      unless confederation_peers.empty?
-        @set_args[:state] = 'no'
-        @set_args[:peer_list] = confederation_peers
-        config_set('bgp', 'confederation_peers', @set_args)
-      end
-      unless peers == default_confederation_peers
-        @set_args[:state] = ''
-        @set_args[:peer_list] = peers
-        config_set('bgp', 'confederation_peers', @set_args)
+      cmds = config_get('bgp', 'confederation_peers', @get_args)
+      return default_confederation_peers if cmds.to_s.empty?
+      cmds = cmds.split(' ') if cmds.instance_of? String
```

Comment:
Yeah, if you set the default_value properly it will be automatically returned by config_get - see below for my suggested approach here.